### DISCUSSION TITLE
r.sim.water: Where does "discharge" come from?

### DISCUSSION THREAD

**Discussion by xycarto** [OSGeo/grass#4534] [General]
Thank you for the patience on the constant questions about this tool.  I am learning the code structure in the hopes of making my own modifications in the future

In the 'r.sim.water` tool, the observation points output "discharge" values to the log file. In the code, the comments suggests this should be "depths". Where/how is "discharge" passed as the observation points. I'd really like to take a crack at modifying this myself to "depths". I am pretty new to C type code, but this is an excellent opportunity for me to have go 

Again, thanks for all the help from this community thus far.

**Comment by petrasovaa** [comment]
The code is here: https://github.com/OSGeo/grass/blob/main/raster/r.sim/simlib/hydro.c#L449
For r.sim.water, `wdepth` is NULL, so the output is computed in the same way as discharge raster here:
https://github.com/OSGeo/grass/blob/main/raster/r.sim/simlib/output.c#L232
Couple lines above, depth is computed as `pow(gama[i][j], 3. / 5.)`.

**Reply by xycarto** [reply]
Once again, thank you @petrasovaa
